solutions/swb-app/src/keyPairs/listKeyPairsRequest.test.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import JSONValue from '@aws/workbench-core-base/lib/types/json';
+import { ListKeyPairsRequestParser } from './listKeyPairsRequest';
+
+describe('ListKeyPairsRequestParser', () => {
+  let requestObject: Record<string, JSONValue>;
+
+  beforeEach(() => {
+    requestObject = { userId: 'valid-uid' };
+  });
+  describe('when pageSize', () => {
```

Comment:
This page size logic seems to be getting spread to a lot of different requests. If you get some time, it would be worth looking into writing a custom Zod transformer that can encapsulate this logic and be shared with all the other requests that use it.

Reply:
We have a backlog item for this I believe!